python/triton/language/semantic.py

```diff
@@ -1153,12 +1153,16 @@ def reinterpret_tensor_descriptor(desc_ptr: tl.tensor, block_ty: tl.block_type,
 
 
 def validate_descriptor_block(shape, dtype):
-    if len(shape) != 2:
+    if len(shape) < 2:
+        return
+    elem_bytes = dtype.primitive_bitwidth / 8
+    contig_dim_size = shape[-1] * elem_bytes
+    if contig_dim_size < 32:
         return
     # Due to limitations of the shared memory encoding, the TMA bounding box has
     # to be at least as big as the swizzle tile.
-    assert shape[0] >= 8, f"tensor descriptor block shape must have at least 8 rows, but got {shape[0]}"
-    min_cols = 32 // dtype.primitive_bitwidth * 8
+    assert shape[-2] >= 8, f"tensor descriptor block shape must have at least 8 rows, but got {shape[0]}"
```

Comment:
Full disclosure: there's a bit of a rough edge here where you can have < 8 rows if there are few enough columns that we can use the unswizzled layout but not if there are more columns. If this becomes an issue I think it's fixable but we have to be careful not to accidentally disable swizzling for gather and scatter which have 1-row descriptors.

Reply:
Is there some way this can be documented in the on the frontend functions?

Reply:
> there's a bit of a rough edge here where you can have < 8 rows if there are few enough columns that we can use the unswizzled layout but not if there are more columns.

if there are more columns can't we emit more tma messages?

Reply:
The problem is that for the swizzled layouts the linear layout code will error out if there are fewer than 8 rows. So our options are:
1. Fix the layout code to not assume 8 rows minimum (might need to test if tma supports this, but I think it should given the gather/scatter messages are only 4 rows at a time)
2. Pick an unswizzled layout when there are < 8 rows, but not if the user is a gather or scatter.

Reply:
> Pick an unswizzled layout when there are < 8 rows, but not if the user is a gather or scatter.

That's for performance right? I thought we would always pick an unswizzled layout by default and let users propagate their layout for performance.

Reply:
Okay, I can go that route and just add some heuristics for when gather/scatter is called outside of the pipeliner.

Reply:
what's different when gather/scatter is called outside of the pipeliner. We want swizzling in those cases based on the users of the gather/scatter? Or is there a reason why gather/scatter prefers swizzle layouts?

Reply:
Sorry, what I meant was when the gather result is used by a dot op then AccelerateMatmul will choose the swizzled layout for the `LocalAllocOp` and the encoding will be propagated from there. However, if it's used more generally then there might not be a `LocalAllocOp` and we do still fall back to the convention.

Reply:
>  However, if it's used more generally then there might not be a LocalAllocOp and we do still fall back to the convention.

I see, I wonder if fall back should be no swizzle? Anyway that's a detail